### PR TITLE
Adds replace_keys function to Arr

### PR DESCRIPTION
**Pull request by sagikazarmark**
Adds a replace_key implementation. It is important the order of the array remains the same without any array slicing.

### CODE REVIEW

**Comment by it-can** [comment]
Please add some tests... :-)

**Comment by sagikazarmark** [comment]
I will, if someone confirms that it will be merged. I will also add docs then. ;)

**Comment by WanWizard** [comment]
A potential problem I see is a key conflict, which causes an overwrite, and a key array that is out of sync with the values.

```
$array = ['a' => 1, 'b' => 2, 'c' => 3];
$array = replace_keys($array, ['a' => 'c', 'c' => 'a']);
```

result?

File: classes/arr.php [hunk 1]
```diff
@@ -1266,4 +1266,30 @@ public static function subset(array $array, array $keys, $default = null)
...
+	 */
+	public static function replace_keys(array $array, array $keys)
+	{
+		$all_keys = array_keys($array);
+
+		// Loop all keys
+		foreach ($all_keys as &$key)
+		{
+			// and replace old ones
+			if (array_key_exists($key, $keys))
```
Comment (sagikazarmark) [inline classes/arr.php:1286]: @WanWizard Correct. Would a check for new key existence solve it?
`array_key_exists($key, $keys) and ! array_key_exists($keys[$key], $keys)`

Comment (WanWizard) [inline classes/arr.php:1286]: If you do that, you get incomplete/incorrect results. Given the fact you have an indexed array to start with, perhaps it's better to create a new array?

```
$new_keys = array();
foreach ($all_keys as &$key)
{
    // and replace old ones
    if (array_key_exists($key, $keys))
    {
        $new_keys[] = $keys[$key];
    }
    else
    {
         $new_keys[] = $key;
    }
}
```

Comment (WanWizard) [inline classes/arr.php:1286]: Scratch that, that fails too, on

```
$array = ['a' => 1, 'b' => 2, 'c' => 3];
$array = replace_keys($array, ['a' => 'c']);
```

**Comment by WanWizard** [comment]
If you can document this with a fat warning that in case of conflicts values are overwritten, and it's the dev's responsibility this doesn't happen, this can be merged.

**Comment by WanWizard** [comment]
@sagikazarmark what to do this one? No tests and docs were ever made?

**Comment by sagikazarmark** [comment]
I don't know. Is it necessary at all? V1 is in feature freeze, isn't it. If you think it can be useful, I can add it to the v2 repo.

**Comment by WanWizard** [comment]
Correct. Just cleaning up old issues for a hopefully final 1.7.3. release. Moved to v2.

**Pull request closed without merging**